src/theia/sfm/feature_extractor_and_matcher.cc

```diff
@@ -224,8 +282,18 @@ void FeatureExtractorAndMatcher::ProcessImage(
   // Extract Features.
   std::vector<Keypoint> keypoints;
   std::vector<Eigen::VectorXf> descriptors;
-  ExtractFeatures(options_, image_filepath, &keypoints, &descriptors);
-
+  if (mask_filepaths_.empty()) {
+    ExtractFeatures(options_,
```

Comment:
When using the FindWithDefault above, you do not need this if statement at all. If the mask does not exist then the logic in ExtractFeatures will handle that.